src/testRunner/compilerRunner.ts

```diff
@@ -51,9 +51,10 @@ export class CompilerBaselineRunner extends RunnerBase {
         return this.testSuiteName;
     }
 
+    private loadedTestFiles: CompilerFileBasedTest[] | undefined;
     public enumerateTestFiles() {
         // see also: `enumerateTestFiles` in tests/webTestServer.ts
-        return this.enumerateFiles(this.basePath, /\.tsx?$/, { recursive: true }).map(CompilerTest.getConfigurations);
+        return this.loadedTestFiles ??= this.enumerateFiles(this.basePath, /\.tsx?$/, { recursive: true }).map(CompilerTest.getConfigurations);

```

Comment:
I think these need to be released at some point as otherwise we are holding onto this data for too long (till all tests are running?)

Reply:
It would be until the process ends I believe, but I don't think that this actually ends up holding on to too much; it's just a list of filenames, right?

Reply:
Its typed :  `CompilerFileBasedTest` so not sure if its just file names or more ?

Reply:
So, it'll include filenames, but also file contents and configurations. Honestly, I'm shocked that we do that stuff up front, but I suppose you need the file contents to know the config to then know what tests to generate...

Reply:
@sheetalkamat @jakebailey The more aggressive approach would be to return just the `string[]` from `enumerateTestFiles`. I do not see an instance where the extra information returned is used. Maybe it was in the past and it is no longer needed? https://github.com/microsoft/TypeScript/commit/fd8b18244bffb349cd1bf54e31e597041422d24e

This would remove the double reading and parsing of the test files and there would not be a need to cache the result.

Reply:
That actually seems great.

Reply:
Pushed the new version.